pkg/provisioningv2/rke2/planner/store.go

```diff
@@ -319,17 +319,15 @@ func (p *PlanStore) UpdatePlan(entry *planEntry, plan plan.NodePlan, maxFailures
 	}
 
 	secret = secret.DeepCopy()
-
 	if secret.Data == nil {
-		secret.Data = map[string][]byte{}
+		// Create the map with enough storage for what is needed.
+		secret.Data = make(map[string][]byte, 6)
 	}
 
 	rke2.CopyPlanMetadataToSecret(secret, entry.Metadata)
 
-	// if there are no probes, clear the statuses of the probes to prevent false positives
-	if len(plan.Probes) == 0 {
-		delete(secret.Data, "probe-statuses")
-	}
+	// If the plan is being updated, then delete the probe-statuses so their healthy status will be reported as healthy only when they pass.
+	delete(secret.Data, "probe-statuses")
```

Comment:
With this change, if the probe status success thresholds are not set to `1`, we will have false positives on failures. Not that this matters right now because I don't think we currently have any probes that have successThreshold > 1 but still, something to keep in mind. https://github.com/rancher/system-agent/blob/v0.2.5/pkg/prober/prober.go#L121-L123